src/libs/migrations/KeyReportActionsByReportActionID.js

```diff
@@ -0,0 +1,59 @@
+import _ from 'underscore';
+import Onyx from 'react-native-onyx';
+import Log from '../Log';
+import ONYXKEYS from '../../ONYXKEYS';
+
+/**
+ * This migration updates reportActions data to be keyed by reportActionID rather than by sequenceNumber.
+ *
+ * @returns {Promise}
+ */
+export default function () {
+    return new Promise((resolve) => {
+        const connectionID = Onyx.connect({
+            key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
+            waitForCollectionCallback: true,
+            callback: (allReportActions) => {
+                Onyx.disconnect(connectionID);
+
+                if (!allReportActions) {
+                    Log.info('[Migrate Onyx] Skipped migration KeyReportActionsByReportActionID');
+                    return resolve();
+                }
+
+                const newReportActions = {};
+                const allReportActionsEntires = Object.entries(allReportActions || {});
+                for (let i = 0; i < allReportActionsEntires.length; i++) {
+                    const [onyxKey, reportActionsForReport] = allReportActionsEntires[i];
+                    if (reportActionsForReport) {
+                        const newReportActionsForReport = {};
+                        const reportActionsForReportEntries = Object.entries(reportActionsForReport);
+                        for (let j = 0; j < reportActionsForReportEntries.length; j++) {
+                            const [reportActionKey, reportAction] = reportActionsForReportEntries[j];
+                            if (!_.isNaN(Number(reportActionKey))
+                                && Number(reportActionKey) === Number(reportAction.reportActionID)
+                                && Number(reportActionKey) !== Number(reportAction.sequenceNumber)) {
+                                Log.info('[Migrate Onyx] Skipped migration KeyReportActionsByReportActionID');
+                                return resolve();
+                            }
```

Comment:
This is saying that we assume if one report is migrated, then all are?
And if so, is that correct?

Reply:
> This is saying that we assume if one report is migrated, then all are?

Correct.

> And if so, is that correct?

I think so. Can you think of any scenario where only some of the reportActions are migrated but not others?

Reply:
Maybe when you did not load the app for a while (so you have old data) but receive actions with the new data from a notification?

Reply:
Oh, push notifications in an interesting case we haven't talked about yet. Going to bring this up in slack

Reply:
Posted in slack https://expensify.slack.com/archives/C04DC6LU2UB/p1674242808964579